packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -20,14 +27,278 @@ import {
   DROPDOWN_CHIP_COLORS,
   DROPDOWN_ID,
   DROPDOWN_MENU_PLACEMENT,
-  DROPDOWN_MENU_POSITION
+  DROPDOWN_MENU_POSITION,
+  DropdownOption
 } from "./DropdownConstants";
 import generateBaseStyles, { customTheme } from "./Dropdown.styles";
 import Control from "./components/Control/Control";
 import menuStyles from "./components/menu/menu.module.scss";
 import styles from "./Dropdown.module.scss";
+import { VibeComponent, VibeComponentProps } from "../../types";
 
-const Dropdown = forwardRef(
+interface CustomSingleValueProps extends SingleValueProps<DropdownOption> {
+  Renderer: React.ComponentType;
+  readOnly: boolean;
+  selectedOption: DropdownOption;
+}
+
+interface CustomMenuProps extends MenuProps<OptionTypeBase, boolean> {
+  dropdownMenuWrapperClassName: string;
+}
+
+interface CustomOptionProps extends OptionProps<OptionTypeBase, boolean> {
+  optionWrapperClassName?: string;
+}
+
+type SelectEvent = {
+  action: "select-option" | "clear";
+  option?: DropdownOption;
+};
+
+type DropdownState = {
+  isDisabled: boolean;
+  selectProps: { withReadOnlyStyle: boolean; readOnly: boolean };
+};
+
+export interface DropdownComponentProps extends VibeComponentProps {
+  /**
+   * Custom style
+   */
+  className?: string;
+  /** ClassName to be added to dropdown option wrapper (dropdown-wrapper__option--reset) */
+  optionWrapperClassName?: string;
+  /** ClassName to be added to dropdown single value wrapper (dropdown-wrapper__single-value--reset) */
+  singleValueWrapperClassName?: string;
+  /** ClassName to be added to dropdown menu wrapper (dropdown-menu-wrapper) */
+  dropdownMenuWrapperClassName?: string;
+  /**
+   * Placeholder to show when no value was selected
+   */
+  placeholder?: string;
+  /**
+   * If set to true, dropdown will be disabled
+   */
+  disabled?: boolean;
+  /**
+   * If set to true, dropdown won't be editable
+   */
+  readOnly?: boolean;
+  /**
+   * Called when menu is opened
+   */
+  onMenuOpen?: (event: React.MouseEvent) => void;
+  /**
+   * Called when menu is closed
+   */
+  onMenuClose?: (event: React.MouseEvent) => void;
+  /**
+   * Called when key is pressed in the dropdown
+   */
+  onKeyDown?: (event: React.MouseEvent) => void;
+  /**
+   * Called when focused
+   */
+  onFocus?: (event: React.MouseEvent) => void;
+  /**
+   * Called when blurred
+   */
+  onBlur?: (event: React.MouseEvent) => void;
+  /**
+   * Called when selected value has changed
+   */
+  onChange?: (
+    option: DropdownOption | DropdownOption[],
+    event: SelectEvent | React.MouseEvent | React.KeyboardEvent
+  ) => void;
+  /**
+   * Called when the dropdown's input changes.
+   */
+  onInputChange?: (event: React.KeyboardEvent) => void;
+  /**
+   * If true, search in options will be enabled
+   */
+  searchable?: boolean;
+  /**
+   * The dropdown options
+   */
+  options?: DropdownOption[];
```

Comment:
@talkor let's monitor if it is breaking existing behaviors
if so:
> Everywhere where we use `DropdownOption`, we should be caution with it
We'll might need to replace to `OptionTypeBase` as a first step, and in the future make the Dropdown component accept a generic for `OptionType` that extends `OptionTypeBase`.